### PR TITLE
Feature/routes protection

### PR DESCRIPTION
**Pull request by ArtsyBiba**
Adding middleware to protect routes

### CODE REVIEW

File: client/src/components/CreateFriendsList/User.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    
+    useEffect(() => {
+        const checkIfAdded = () => {
+            return (
+                setAdded(newFriendsList.friends.some(e => e._id === user._id))
+            )
+        };
+
+        checkIfAdded();
+    });
```
Comment (bonnieli) [inline client/src/components/CreateFriendsList/User.js:20]: missing an empty dependency here?

File: client/src/components/CreateFriendsList/index.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+        await createFriendsList(newFriendsList);
+        setReload(!reload);
+        
+        setNewFriendsList(initialFriendsList);
+        setOpenCreateFriendsList(false);
+    };
+    
+    const createFriendsList = async (newFriendsList) => {
+        let token = localStorage.getItem('auth-token');
```
Comment (bonnieli) [inline client/src/components/CreateFriendsList/index.js:55]:
```suggestion
        const token = localStorage.getItem('auth-token');
```

File: server/routes/friendRouter.js [hunk 3]
```diff
@@ -1,27 +1,29 @@
 const router = require('express').Router();
 const User = require('../models/userModel');
+const auth = require('../middleware/auth');
 
-router.get('/usersList', async (req, res) => {
+router.get('/', auth, async (req, res) => {
     try {
-        const user = await User.find({});
+        const users = await User.find({});
+        const filteredUsers = users.filter((user) => user._id != req.user);
```
Comment (bonnieli) [inline server/routes/friendRouter.js:8]: instead of filtering it manually, take advantage of your db queries to filter our the users

**Review by bonnieli — APPROVED** [review]

File: client/src/components/CreateFriendsList/User.js (hunk 1 above)
Comment (ArtsyBiba) [reply client/src/components/CreateFriendsList/User.js:20]: yep, updated

File: client/src/components/CreateFriendsList/index.js (hunk 2 above)
Comment (ArtsyBiba) [reply client/src/components/CreateFriendsList/index.js:55]: changed

File: server/routes/friendRouter.js (hunk 3 above)
Comment (ArtsyBiba) [reply server/routes/friendRouter.js:8]: Refactored with User.find({ '_id': { '$ne': req.user } })

**Pull request closed without merging**